server/src/formatting.ts

```diff
@@ -3,56 +3,76 @@ import {
 import { execFileSync } from 'child_process';
 import { getPerlAssetsPath } from "./assets";
 
 
 export function formatDoc(params: DocumentFormattingParams, txtDoc: TextDocument, settings: NavigatorSettings): TextEdit[] | undefined {
+    const start = Position.create( 0,0 );
+    const end = Position.create( txtDoc.lineCount, 0 );
+    const range = {start, end} as Range;
     const text = txtDoc.getText();
-    const fixedSource = perltidy(text, settings);
-    const start = { line: 0, character: 0 };
-    const end = { line: txtDoc.lineCount, character: 0 };
-    const range = {start, end};
-    if(fixedSource){
-        let edits: TextEdit = {
-            range: range,
-            newText: fixedSource
-        };
-        return [edits];
-    } else {
-        return;
-    }
-} 
 
-export function formatRange(params: DocumentRangeFormattingParams, txtDoc: TextDocument, settings: NavigatorSettings): TextEdit[] | undefined {
+    return maybeReturnEdits(range, txtDoc, settings);
+}
 
-    const start = { line: params.range.start.line, character: 0 };
+export function formatRange(params: DocumentRangeFormattingParams, txtDoc: TextDocument, settings: NavigatorSettings): TextEdit[] | undefined {
+    const start = Position.create( params.range.start.line, 0 );
     const offset = params.range.end.character > 0 ? 1 : 0;
-    const end = { line: params.range.end.line + offset, character: 0 };
-    const range = {start, end};
+    const end = Position.create(params.range.end.line + offset, 0 );
+    const range = {start, end} as Range;
     const text = txtDoc.getText(range);
 
-    const fixedSource = perltidy(text, settings);
+    return maybeReturnEdits(range, txtDoc, settings);
+}
 
-    if(fixedSource){
-        let edits: TextEdit = {
-            range: range,
-            newText: fixedSource
-        };
-        return [edits];
-    } else {
+function maybeReturnEdits (range: Range, txtDoc: TextDocument, settings: NavigatorSettings): TextEdit[] | undefined {
+    const text = txtDoc.getText(range);
+    if ( !text) {
         return;
     }
+
+    let formattedSource = perltidy(text, settings);
+    formattedSource = perlimports(txtDoc, formattedSource || text, settings);
+
+    if (!formattedSource) {
+        return;
+    }
+
+    let edits: TextEdit = {
+        range: range,
+        newText: formattedSource
+    };
+    return [edits];
 }
 
+function perlimports(doc: TextDocument, code: string, settings: NavigatorSettings): string | undefined {
+    if(!settings.perlimportsTidyEnabled) return;
+    const importsPath = join(getPerlAssetsPath(), 'perlimportsWrapper.pl');
+    let cliParams: string[] = [importsPath].concat(getPerlimportsProfile(settings));
+    cliParams = cliParams.concat(['--read-stdin', '--filename', Uri.parse(doc.uri).fsPath]);
+
+    let output: string | Buffer;
+    try {
+        output = execFileSync(settings.perlPath, cliParams, {timeout: 25000, input: code});
+        output = output.toString();
```

Comment:
I think this could be simplified slightly:

```suggestion
    try {
        const output = execFileSync(settings.perlPath, cliParams, {timeout: 25000, input: code}).toString();
```